includes/admin/class-amp-post-meta-box.php

```diff
@@ -422,4 +429,101 @@ public function preview_post_link( $link ) {
 
 		return $link;
 	}
+
+	/**
+	 * Add a REST API field to display the AMP status on AMP supported post types.
+	 *
+	 * @return void
+	 */
+	public function add_rest_api_fields() {
+		$post_type       = 'post';
+		$should_register = (
+			in_array( $post_type, get_post_types_by_support( 'amp' ), true ) &&
+			in_array( $post_type, get_post_types( [ 'show_in_rest' => true ] ), true )
+		);
+
+		if ( ! $should_register ) {
+			return;
+		}
+
+		register_rest_field(
+			$post_type,
+			self::REST_ATTRIBUTE_NAME,
+			[
+				'get_callback'    => [ $this, 'amp_status_get_callback' ],
+				'update_callback' => [ $this, 'amp_status_update_callback' ],
+				'schema'          => [
+					'description' => __( 'AMP status', 'amp' ),
+					'type'        => 'boolean',
+				],
+			]
+		);
+	}
+
+	/**
+	 * Display the AMP status of a post via the REST API.
+	 *
+	 * @param array $post_data Post data.
+	 * @return bool Whether AMP is enabled on post.
+	 */
+	public function amp_status_get_callback( $post_data ) {
+		$status = $this->sanitize_status( get_post_meta( $post_data['id'], self::STATUS_POST_META_KEY, true ) );
+
+		if ( '' === $status ) {
+			$post              = get_post( $post_data['id'] );
+			$status_and_errors = self::get_status_and_errors( $post );
+
+			if ( isset( $status_and_errors['status'] ) ) {
+				$status = $status_and_errors['status'];
+			}
+		}
+
+		return self::ENABLED_STATUS === $status;
+	}
+
+	/**
+	 * Update the AMP status of a post via the REST API.
+	 *
+	 * @param bool    $is_enabled Whether AMP is enabled.
+	 * @param WP_Post $post       Post being updated.
+	 * @return void|WP_Error Returns an instance of `WP_Error` if the post failed to be updated.
+	 */
+	public function amp_status_update_callback( $is_enabled, $post ) {
+		if ( 'post' !== $post->post_type ) {
```

Comment:
Is this right? Shouldn't rather be checking if the post type is among `get_post_types_for_rest_api`? Otherwise, wouldn't this fail for `page` posts?

Reply:
Is this only supposed to register the REST field for `page` and `post` post types?

In https://github.com/ampproject/amp-wp/pull/4689/commits/91a5eb8d6014c8e85cf3047ed44dc4ff82bdb782 I've reverted to using `AMP_Post_Type_Support::get_post_types_for_rest_api()` to get the supported post types.

Reply:
My understanding is that the parameter for `register_rest_field()` is the object type not the post type (object subtype?). In other words, it could be `post`, `user`, `comment`, etc as opposed to types of post (i.e. the post types). It's confusing because `post` is overloaded being both an object type and a post type. Nevertheless it seemed to work before? So I'm confused.

Reply:
Oh, I see what you mean now. Yes the `object_type` parameter is overloaded and the name is indeed confusing. In fact, the issue has been raised before the REST API was merged into WordPress (see https://github.com/WP-API/WP-API/issues/2692).

Reply:
I shall make sure to never to `register_post_type( 'comment' )`!